web-console/3-project-details/labels.md

```diff
@@ -0,0 +1,104 @@
+# Labels on Detail Pages
+
+## View Labels
+- Labels are currently displayed at the top of most detail pages (Deployments, Pods, Builds, Images, Storage, etc)
+- A new `Labels` section should be added to the bottom of the `Details` tab on all of these pages to allow for a more comprehensive labels area
+
+![Label 1](img/new-labels-00-view.png)
+
+## Add Labels
+![Label 1](img/new-labels-01-add.png)
+
+- A `Manage Labels` link should be displayed below the applied labels
+- Clicking the link will open a modal, displaying the labels that are already applied, as well as an option to add new labels.
+- Clicking `Add` will display a new row with two empty text fields for label Name and label Value.
+
+![Label 1](img/new-labels-02-add.png)
+
+- The save button should not be enabled until valid entries are made.
+
+![Label 1](img/new-labels-03-add.png)
+
+- Invalid characters will result in a field level error once the user changes focus away from the particular field.
+
+![Label 1](img/new-labels-03B-add.png)
+
+- Label names cannot have more than one value associated.
+- Attempting to enter a label name that has already been applied with a value will result in a field level error as well.
+
+![Label 1](img/new-labels-04-add.png)
+
+![Label 1](img/new-labels-05-add.png)
+
+- Once the user clicks `Save` the new label should appear at the bottom of the list of labels on the Details tab.
+
+## Edit Labels
+![Label 1](img/new-labels-01-edit.png)
+
+- The same `Manage Labels` link will be used for editing existing labels.
+- Users can change a name or value as desired, as long as the name is not already associated with another value.
+
+#### Implementation Details
+- Values can be edited in this view, but only the current object will receive the label change. In order to bulk change labels for multiple items (Deployments in this case), users must go to the summary page and multi-select items.
+- If there are any restricted labels that users are unable to change, the entire row should be disabled.
```

Comment:
I wonder if we can be smart about knowing what labels should be editable and which shouldn't.

Reply:
@ncameronbritt you mentioned labels for Pods being read-only so I would think this disabled visual would be used for cases like that. Does that make sense?

Reply:
I think for pods none of the labels would be editable, but your mock shows one that's editable and one that's not. 

What I thought you were saying is that labels like `openshift.io/blahblah` would not be editable, but labels like `app` would be. If that's the case, I was just saying that it would be nice if we could be smart and figure out which should be editable, and which a user shouldn't mess with. Just ignore if I misunderstood what you were saying.

Reply:
Yeah that would be ideal so I left it in the design but for now @spadgett said there is probably no way to do that :)

Reply:
Figured as much. :-)